docs/blog/2021-09-07-mocking-a-production-api.md

```diff
@@ -0,0 +1,300 @@
+"verb": "sale_pending"
+},
+{
+"arguments": "incentivize scalable mindshare",
+"body": null,
+"created_at": "2018-02-21T12:51:36",
+"description": "Received new order",
+"id": 599084,
+"message": "Received new order",
+"path": "/admin/orders/01984/transactions/#3595",
+"subject_id": 1050540,
+"subject_type": "Order",
+"verb": "placed"
+}
+...
+```
+
+We then dumped it all in a MongoDB collection with the one-liner:
+
+```bash
+$ synth generate shopify --to mongodb://localhost:27017/shopify --size 40000
+```
+
+Next step is to re-create the Events API endpoint.
+
+### 2. Creating the API
+
+@brokad
+
+## Day 5: Packaging
+
+// TODO add a picture of Docker here
+
+The data is ready, the API is ready, time to package this thing up and give it
+to people to actually use. Let's see if our experiment was a success.
+
+When thinking about distributing our API, we're optimising for two things:
+
+1. Ease of use - how simple it is for someone to download this thing and
+   get going
+2. Time - we have 2-3 hours to make sure this thing is packaged and ready to go
+
+in that order.
+
+We need to pack the data, database and a NodeJS runtime to actually run the
+server. Our initial idea was to use `docker-compose` with 2 services, the
+database, web-server and the network plumbing to get it to work.. After
+discussing this for a few minutes, we decided that `docker-compose` may be an
+off-ramp for some users as they don't have it or are not familiar with how it
+works. This went against our tenet which is 'ease of use'.  
+
+So we decided to take the slightly harder and hackier route of packaging the 
+whole thing in a single Docker container. It seemed like the best trade-off 
+between goals 1 and 2.
+
+There were 6 steps to getting this thing over the line:
+1. Start with the MongoDB base image. This gives us a Linux environment and 
+   a database.
+2. Download and install NodeJS runtime in the container.
+3. Download and install Synth in the container.
+4. Copy the javacript sources over & the Synth data model
+5. Write a small [ENTRYPOINT](TODO) [shell script](TODO_PATH_TO_OUR_SCRIPT) 
+   to start the `mongod`, server and generate data into the server
+6. Expose port 3000
+
+And we're done! We've ~~hackily~~ happilly packaged our mock API in a 
+platform agnostic one liner.
+
+## Was it a success?
+
+An important aspect of this experiment was to see if we could conceive, 
+research, design and implement a PoC in a week (on the side, we were working 
+on Synth at the same time). And I can safely say this was a success! We got 
+it done to spec. An interesting thing to note is that **60%** of the time was 
+spent on ideating, researching and planning - and only 40% of the time on the actual 
+implementation. However spending all that time before writing code 
+definitely saved a bunch of time, and if we didn't the project would have 
+overshot or failed.
+
+Now if the PoC itself was a success is a different question. This is where 
```

Comment:
Maybe you can elaborate on this? Give a few one-liners, we push the image to GCP or dockerhub and people can just pull and get started?

Also note the data model is overly simplistic and likely needs tweaking and improvements so it'd be nice to point people in the direction of the repo to get some help improving it.